### PR TITLE
fix(js_parser): allow a trailing comma in setters

### PR DESCRIPTION
**Pull request by fireairforce**
## Summary

closes: #4317 

i add a new syntax at `js.ungram` to solve the problem: 

```ts
JsSetterObjectMember =
	'set'
	name: AnyJsObjectMemberName
	'('
	- parameter: AnyJsFormalParameter
	+ parameter: AnyJsFormalParameter','?
	')'
	body: JsFunctionBody
```

Except `JsSetterObjectMember`, i also support `JsSetterClassMember`、`TsSetterSignatureClassMember`, and `TsSetterSignatureTypeMember` here.

## Test Plan

I add test case for `biome_js_parser` and update the formatter test case.

### CODE REVIEW

**Comment by fireairforce** [comment]
anyone is free for review this? thanks~

**Review by Conaclos — COMMENTED** [review]
Thanks for the fix. I left some comments.

File: xtask/codegen/js.ungram [hunk 1]
```diff
@@ -561,7 +561,7 @@ JsSetterObjectMember =
 	'set'
 	name: AnyJsObjectMemberName
 	'('
-	parameter: AnyJsFormalParameter
+	parameter: AnyJsFormalParameter','?
```
Comment (Conaclos) [inline xtask/codegen/js.ungram:564]: I think we should also allow trailing comma for `JsSetterClassMember`, `TsSetterSignatureClassMember`, and `TsSetterSignatureTypeMember`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -74,14 +74,18 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
...
 
-#### New features
-
-- Add support for parsing the defer attribute in import statements ([#4215](https://github.com/biomejs/biome/issues/4215)).
+#### Bug fixes
 
-   ```js
-   import defer * as myModule from "my-module";
-   ```
+- Fix [#4317](https://github.com/biomejs/biome/issues/4317), get accessors parameter can contain a trailing comma, the following example can be parsed correctly:
```
Comment (Conaclos) [inline CHANGELOG.md:79]:
```suggestion
- Fix [#4317](https://github.com/biomejs/biome/issues/4317), setter parameter can contain a trailing comma, the following example will now be parsed correctly:
```

File: xtask/codegen/js.ungram (hunk 1 above)
Comment (fireairforce) [reply xtask/codegen/js.ungram:564]: I think you're right, I didn't take that into account.😂, i will update this later~

File: CHANGELOG.md [hunk 3]
```diff
@@ -74,14 +74,18 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 
 ### Parser
 
-#### New features
-
-- Add support for parsing the defer attribute in import statements ([#4215](https://github.com/biomejs/biome/issues/4215)).
```
Comment (fireairforce) [inline CHANGELOG.md:79]: this pr finally merged again？

Comment (Conaclos) [reply CHANGELOG.md:79]: I don't remember merging a new PR about this. Maybe I am wrong.

Comment (fireairforce) [reply CHANGELOG.md:79]: seems merged by other pr, i will delete here.

File: CHANGELOG.md (hunk 2 above)
Comment (fireairforce) [reply CHANGELOG.md]: updated~

File: xtask/codegen/js.ungram (hunk 1 above)
Comment (fireairforce) [reply xtask/codegen/js.ungram:564]: i updated here and support both `JsSetterClassMember`, `TsSetterSignatureClassMember`, and `TsSetterSignatureTypeMember` and add test case for these, feel free to review it~

**Review by Conaclos — APPROVED** [review]
Thanks!

**Pull request merged**